lsp/src/tree_sitter.rs

```diff
@@ -57,50 +57,58 @@ fn create_attribute(node: Node<'_>, source: &str) -> Option<Position> {
     return None;
 }
 
-fn get_position_by_query(query: &str, node: Node<'_>, source: &str) -> Option<Position> {
+fn get_position_by_query(
+    query: &str,
+    node: Node<'_>,
+    source: &str,
+    trigger_point: Point,
+) -> Option<Position> {
+    // See: https://tree-sitter.github.io/tree-sitter/using-parsers#query-syntax
     let query = Query::new(tree_sitter_html::language(), query)
         .expect(&format!("get_position_by_query invalid query {query}"));
-    let mut cursor = QueryCursor::new();
-    let matches = cursor.matches(&query, node, source.as_bytes());
+    let mut cursor_qry = QueryCursor::new();
+    let matches = cursor_qry.matches(&query, node, source.as_bytes());
 
     debug!("get_position_by_query node {:?}", node.to_sexp());
 
-    // FIXME this only covers suggestion for attributes at the end of the tag!
-    let first_match = matches.last()?;
-
     let capture_names = query.capture_names();
-    debug!("get_position_by_query captures {:?}", capture_names);
-
-    debug!("get_position_by_query first_match {:?}", first_match);
-
-    let props = first_match
-        .captures
-        .iter()
-        .fold(HashMap::new(), |mut props, capture| {
-            let name = capture_names[capture.index as usize].to_owned();
-            let value = capture
-                .node
-                .utf8_text(source.as_bytes())
-                .expect(&format!("failed to parse capture value for '{name}'"))
-                .to_owned();
-
-            props.insert(name, value);
+    debug!("capture_names: {:?}", capture_names);
+
+    let props = matches
+        .into_iter()
+        .fold(HashMap::new(), |mut props, match_| {
+            match_
+                .captures
+                .iter()
+                .filter(|capture| capture.node.start_position() <= trigger_point)
+                .for_each(|capture| {
+                    let name = capture_names[capture.index as usize].to_owned();
+                    let value = capture
+                        .node
+                        .utf8_text(source.as_bytes())
+                        .expect(&format!("failed to parse capture value for '{name}'"))
+                        .to_owned();
+                    props.insert(name, value);
+                });
 
             props
         });
 
+    debug!("props: {:?}", props);
+
     let name = props.get("attr_name")?.to_owned();
 
-    let attr_value = match props.get("attr_value").or(props.get("quoted_attr_value")) {
+    let attr_value = match props.get("quoted_attr_value").or(props.get("attr_value")) {
```

Comment:
There might be a better way to do this, so if there is a "quoted_attr_value" but no "attr_value" the lsp needs to suggest values, but there might be cases where there is attr_value but is just a mismatch of number of quotes 😓